src/ir.cpp

```diff
@@ -28855,7 +28855,33 @@ static IrInstGen *ir_analyze_instruction_check_switch_prongs(IrAnalyze *ira,
         ir_add_error(ira, &instruction->base.base,
             buf_sprintf("else prong required when switching on type '%s'", buf_ptr(&switch_type->name)));
         return ira->codegen->invalid_inst_gen;
-    }
+    } else if(switch_type->id == ZigTypeIdMetaType) {
+        ZigList<ZigValue*> prevs = {0};
```

Comment:
You should be able to use a `HashMap<const ZigType *, ZigValue *, type_ptr_hash, type_ptr_eql>` here.

Reply:
I fixed some things since you reviewed, namely fixing a memory leak and only storing a pointer to the ZigType. But I still think a simple ZigList is more efficient than a hash map in this case. 
* The internal_get function of HashMap does the same thing as the code in the PR but with more steps. 
* The hash map stores both a key and a value, we only need the key. 
* We know the maximum memory footprint at the beginning of scope but HashMap allocates more memory at 60% capacity.
* Insertion into the hash map will also be slower than the array (see HashMap::internal_put).

But I might be wrong, I haven't benchmarked the two approaches.

Reply:
It doesn't do the same. The `HashMap` only loops if the bucket is filled with a different key, while the list version loops over all precious values for each prong. The extra memory usage is negligible as long as the map is freed begore returning.

If you look at the duplicate check for enums you'll see that we use a map there. I doubt it'll actually have a noticeable difference in any real-world situation, but generally problems like this are better solved with hashmaps/sets.

Reply:
Sorry, I somehow totally missed what internal_get was doing. I do agree that a set is probably the right data structure to use here.

Reply:
Sorry, I somehow totally missed what internal_get was doing. Will change it to use a set instead.